### PR TITLE
support volume in grid cli

### PR DESCRIPTION
**Pull request by SalmaElsoly**
### Description

Describe the changes introduced by this PR and what does it affect

### Changes

List of changes this PR includes
- grid cli
  - add flag volume for vm deployment
  - edit deploy.go to include volume
  -  add volume in building the filter
### Testing
- deployed a vm with volume, mounts include volume
### Related Issues

List of related issues
https://github.com/threefoldtech/tfgrid-sdk-go/issues/1140#issue-2439552576
### Checklist

- [x] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstring

### CODE REVIEW

File: grid-cli/cmd/deploy_vm.go [hunk 1]
```diff
@@ -204,6 +214,7 @@ func init() {
 	deployVMCmd.Flags().Int("disk", 0, "disk size in gb mounted on /data")
 	deployVMCmd.Flags().String("flist", ubuntuFlist, "flist for vm")
 	deployVMCmd.Flags().StringSlice("gpus", []string{}, "gpus for vm")
+	deployVMCmd.Flags().Int("volume", 0, "volume size in gb")
```
Comment (Eslam-Nawara) [inline grid-cli/cmd/deploy_vm.go]: mounted on /volume

File: grid-cli/cmd/deploy_vm.go [hunk 2]
```diff
@@ -132,6 +136,12 @@ var deployVMCmd = &cobra.Command{
 			mount = workloads.Disk{Name: diskName, SizeGB: disk}
```
Comment (Eslam-Nawara) [inline grid-cli/cmd/deploy_vm.go]: rename to diskMount

File: grid-cli/internal/cmd/deploy.go [hunk 3]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 // DeployVM deploys a vm with mounts
-func DeployVM(ctx context.Context, t deployer.TFPluginClient, vm workloads.VM, mount workloads.Disk, node uint32) (workloads.VM, error) {
+func DeployVM(ctx context.Context, t deployer.TFPluginClient, vm workloads.VM, mount workloads.Disk, volumeMount workloads.Volume, node uint32) (workloads.VM, error) {
```
Comment (Eslam-Nawara) [inline grid-cli/internal/cmd/deploy.go]: diskMount

**Review by Eslam-Nawara — CHANGES_REQUESTED** [review]

File: grid-cli/cmd/deploy_vm.go [hunk 4]
```diff
@@ -143,7 +153,7 @@ var deployVMCmd = &cobra.Command{
 		}
 
 		if node == 0 {
-			filter, disks, rootfss := filters.BuildVMFilter(vm, mount, farm)
+			filter, disks, rootfss := filters.BuildVMFilter(vm, diskMount, farm)
```
Comment (Eslam-Nawara) [inline grid-cli/cmd/deploy_vm.go]: the volume mount should also be accounted for in building the filter

**Review by Eslam-Nawara — CHANGES_REQUESTED** [review]

File: grid-cli/cmd/deploy_vm.go (hunk 4 above)
Comment (SalmaElsoly) [reply grid-cli/cmd/deploy_vm.go:156]: does volume will be passed to filterNodes( )? , to know if i have to return it from BuildVmFilter or not

**Review by rawdaGastan — APPROVED** [review]

**Review by Eslam-Nawara — APPROVED** [review]

**Pull request merged**